src/wp-admin/includes/upgrade.php

```diff
@@ -2275,6 +2279,35 @@ function upgrade_560() {
 		save_mod_rewrite_rules();
 	}
 }
+/**
+ * Executes changes made in WordPress 5.7.0.
+ *
+ * @ignore
+ * @since 5.7.0
+ */
+function upgrade_570() {
+	global $wp_current_db_version, $wpdb;
+
+	if ( $wp_current_db_version < 51769 ) {
+		/*
+		 * Create fulltext indexes for `wp_posts`.
+		 */
+		$result = $wpdb->query( "ALTER TABLE $wpdb->posts ADD FULLTEXT INDEX `wp_posts_fulltext` (`post_title` ASC, `post_excerpt` ASC, `post_content` ASC);" );
```

Comment:
A nicer way of detecting this that can be added to the `$wpdb->has_cap()` method would be dandy.